crypto/src/swap.rs

```diff
@@ -1,13 +1,43 @@
 use crate::ka;
 use crate::transaction::Fee;
 use anyhow::Result;
+use chacha20poly1305::{
+    aead::{Aead, NewAead},
+    ChaCha20Poly1305, Key, Nonce,
+};
+use decaf377::FieldExt;
+use once_cell::sync::Lazy;
 use penumbra_proto::{dex as pb, Protobuf};
 
 use crate::asset::Id as AssetId;
 use crate::keys::OutgoingViewingKey;
 
-// TODO: not sure what byte size is necessary here yet
-pub const SWAP_CIPHERTEXT_BYTES: usize = 128;
+// Swap ciphertext byte length
+pub const SWAP_CIPHERTEXT_BYTES: usize = 169;
+// Swap plaintext byte length
+pub const SWAP_LEN_BYTES: usize = 153;
+pub const OVK_WRAPPED_LEN_BYTES: usize = 80;
+
+/// The nonce used for swap encryption.
+///
+/// The nonce will always be `[0u8; 12]` which is okay since we use a new
+/// ephemeral key each time.
+pub static SWAP_ENCRYPTION_NONCE: Lazy<[u8; 12]> = Lazy::new(|| [0u8; 12]);
+
+// Can add to this/make this an enum when we add additional types of swaps.
+// TODO: is this actually something we would do? suppose it doesn't hurt to build this
+// in early.
+pub const SWAP_TYPE: u8 = 0;
```

Comment:
My feeling is that we probably don't want to build this in (at least, in this place), and that, in keeping with the "well-oiled joint principle", it'd be better to have that flexibility at the level of the transaction `Actions`, and handle changes by defining new actions. 

If we did it just at the level of the swap data, we'd only be able to make changes subject to the constraints that the new swap types fit exactly within the existing `Swap` action, which is already quite constraining.

So I think we should drop the type parameter here and just hardcode the binary format (then migrate off it if/when we need to by making a new action).